### PR TITLE
Fix/add socketio rooms

### PR DESCRIPTION
**Pull request by Elon-Abulafia**

### CODE REVIEW

File: pages/api/socketio.js [hunk 1]
```diff
@@ -29,7 +33,7 @@ const ioHandler = (req, res) => {
             socketMessage = extractImageFromInputOutputMessage(msg);
           }
 
-          socket.broadcast.emit(emitName, socketMessage)
+          socket.to(msg["source"]).emit(logType, socketMessage)
```
Comment (tomerdayan168) [inline pages/api/socketio.js:40]: Are you sure that this is not happening when the emit name is only log type? ->  Let's say we have two routines - A, B. Both of them log input and output. A user is connecting with two tabs, which means two sockets are opening and two routines are sending input/output. The emit message is the same so both of them get both input/output of both routine.

**Review by tomerdayan168 — COMMENTED** [review]
:goat:

File: pages/api/socketio.js (hunk 1 above)
Comment (San-Moshe) [reply pages/api/socketio.js:40]: I think that specifying room name `to(msg["source)` handles your described use-case.
It should be tested.

File: pages/main-page-view.js [hunk 2]
```diff
@@ -7,9 +7,10 @@ import RoutineExpandView from "../components/routine-expand-view";
 import Divider from "@mui/material/Divider";
 import { getRoutines } from "../utils/api_calls";
 import io from 'socket.io-client'
+import { SERVER_URL } from '../config';
 
 export default function MainPageView() {
-  fetch("/api/socketio").finally(() => {
+  fetch(`${SERVER_URL}/api/socketio`).finally(() => {
```
Comment (San-Moshe) [inline pages/main-page-view.js]: why finally over then?

**Review by San-Moshe — CHANGES_REQUESTED** [review]

**Review by tomerdayan168 — APPROVED** [review]

**Review by San-Moshe — APPROVED** [review]

**Pull request merged**